src/test/java/md/ramaiana/foodmarket/service/GoodServiceTest.java

```diff
@@ -0,0 +1,313 @@
+package md.ramaiana.foodmarket.service;
+
+import md.ramaiana.foodmarket.dao.GoodDao;
+import md.ramaiana.foodmarket.dao.GoodGroupDao;
+import md.ramaiana.foodmarket.model.Brand;
+import md.ramaiana.foodmarket.model.Good;
+import md.ramaiana.foodmarket.model.GoodGroup;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+class GoodServiceTest {
+    @Mock
+    private GoodDao goodDaoMock;
+
+    @Mock
+    private GoodGroupDao goodGroupDaoMock;
+
+    @InjectMocks
+    private GoodService goodService;
+
+    @Test
+    void test_findGoodsFiltered_allParams_returnedList() {
+        //ARRANGE
+        Integer brandId = 2;
+        Integer groupId = 1;
+        String goodName = "someGoodName";
+        Brand someBrand = Brand.builder()
+                .id(brandId)
+                .name("someName")
+                .build();
+        GoodGroup someGroup = GoodGroup.builder()
+                .id(groupId)
+                .name("someName")
+                .build();
+        List<Good> goods = new ArrayList<>();
+        goods.add(Good.builder()
+                .id(3)
+                .name(goodName)
+                .brandId(someBrand.getId())
+                .groupId(someGroup.getId())
+                .price(15f)
+                .build());
+        when(goodDaoMock.getAllByGroupIdAndBrandIdAndNameLike(eq(groupId), eq(brandId), eq(goodName)))
```

Comment:
Could be extracted in private method, like `givenGoods()`, and used in all methods below.